### PR TITLE
Enhancement/add yup

### PR DESCRIPTION
**Pull request by eugenialcaraz**
#### Trello board reference:

* [12](Add yup as form validation)

---

#### Description:

* Add yup as form validator
* Add custom hook to return errors if the validation is incorrect
* Add validation schemas for every form
* Adjust error message to show just the 1st error 

---

#### Reviewers:

* @nicmalegre 
* @jpirazusta 
* @tobicorradi 
* @diego317 

---

### CODE REVIEW

File: src/components/common/forms/validationSchema.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import * as yup from "yup";
+
+const required = "All fields are required";
+const contactFormRequired = "Both fields are mandatory";
+
+export const signInValidationSchema = yup
+  .object({
+    email: yup
+      .string()
```
Comment (nicmalegre) [inline src/components/common/forms/validationSchema.tsx:9]: Did you consider to use `email()` here? Like [this](https://github.com/jquense/yup#getting-started)

File: src/components/common/forms/validationSchema.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      .max(64, "Please enter a valid email")
+      .required(required),
+    password: yup.string().required(required),
+  })
+  .required(required);
+
+export const signUpValidationSchema = yup
+  .object({
+    first_name: yup.string().required(required),
+    email: yup
```
Comment (nicmalegre) [inline src/components/common/forms/validationSchema.tsx:20]: Same here

File: src/components/common/forms/validationSchema.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      .string()
+      .max(64, "Please enter a valid email")
+      .required(required),
+    password: yup
+      .string()
+      .min(6, "Password should be at least 6 characters long")
+      .required(required),
+    password_confirmation: yup
```
Comment (nicmalegre) [inline src/components/common/forms/validationSchema.tsx:29]: Maybe here would be nice to have a validation to check if both passwords are the same. Yup have something called [ref](https://github.com/jquense/yup#refpath-string-options--contextprefix-string--ref). If you want to try:  

`validationSchema: Yup.object({
  password: Yup.string().required('Password is required'),
  passwordConfirmation: Yup.string()
     .oneOf([Yup.ref('password'), null], 'Passwords must match')
});`

**Review by nicmalegre — APPROVED** [review]

File: src/components/common/forms/ContactForm.tsx [hunk 4]
```diff
@@ -21,14 +24,13 @@ const ContactForm = () => {
   };
 
   const isFormValid = !isSubmitted || isValid;
+  const errorMessage = String(Object.values(errors)[0]?.message || "");
```
Comment (jpirazusta) [inline src/components/common/forms/ContactForm.tsx]:
```suggestion
  const errorMessage = String(Object.values(errors)[0]?.message ?? "");
```
You can use the `??` operator in these cases

File: src/components/common/forms/Forms.module.css [hunk 5]
```diff
@@ -56,3 +56,13 @@
...
+
+.signUpForm > div > input,
+.signUpForm > div > select {
+  max-width: 18rem;
+}
+
+.form > div > input,
+.form > div > select {
+  max-width: 18.8rem;
+}
```
Comment (jpirazusta) [inline src/components/common/forms/Forms.module.css]: You could use some classes instead

File: src/components/common/forms/index.tsx [hunk 6]
```diff
@@ -1,3 +1,4 @@
 export * from "./SignInForm";
 export * from "./SignUpForm";
 export * from "./ContactForm";
+export * from "./validationSchema";
```
Comment (jpirazusta) [inline src/components/common/forms/index.tsx:4]: I'd put this inside another folder, not in `components`

**Review by jpirazusta — APPROVED** [review]

File: src/components/common/forms/Forms.module.css (hunk 5 above)
Comment (eugenialcaraz) [reply src/components/common/forms/Forms.module.css]: Fixed!

File: src/components/common/forms/ContactForm.tsx (hunk 4 above)
Comment (eugenialcaraz) [reply src/components/common/forms/ContactForm.tsx]: Ohh, I didn't know. Thanks! Updated

File: src/components/common/forms/validationSchema.tsx (hunk 3 above)
Comment (eugenialcaraz) [reply src/components/common/forms/validationSchema.tsx:29]: I wasn't aware of the ref, I love it! Updated

File: src/components/common/forms/validationSchema.tsx (hunk 2 above)
Comment (eugenialcaraz) [reply src/components/common/forms/validationSchema.tsx:20]: Updated!

File: src/components/common/forms/validationSchema.tsx (hunk 1 above)
Comment (eugenialcaraz) [reply src/components/common/forms/validationSchema.tsx:9]: I agree! Updated!

**Pull request merged**